ops/reth-cross-test/phases/base.py

```
@@ -0,0 +1,425 @@
+    genesis = json.loads(genesis_path.read_text())
+    morph_config = genesis.get("config", {}).get("morph", {})
+
+    # Skip if already patched
+    if morph_config.get("genesisStateRoot"):
+        print("  Genesis already patched with ZkTrie state root.")
+        return
+
+    # Query geth for the genesis block to get ZkTrie state root and hash
+    # Start a temporary geth container to get the correct values
+    print("  Querying geth for ZkTrie genesis state root...")
+    result = subprocess.run(
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)